From 1995d2afc449c2e1515113eabb0ce07c6608a35e Mon Sep 17 00:00:00 2001
From: Violet Purcell <vimproved@inventati.org>
Date: Sat, 4 Nov 2023 12:09:20 -0400
Subject: [PATCH] elf.h: add typedefs for Elf64_Relr and Elf32_Relr

When DT_RELR support was added, it seems that the typedefs for
Elf64_Relr and Elf32_Relr weren't created. This patch fixes that, using
the same typedefs as elfutils 0.190 and glibc.

Signed-off-by: Violet Purcell <vimproved@inventati.org>
--- a/include/elf.h
+++ b/include/elf.h
@@ -558,6 +558,11 @@ typedef struct {
 
 
 
+typedef Elf32_Word Elf32_Relr;
+typedef Elf64_Xword Elf64_Relr;
+
+
+
 #define ELF32_R_SYM(val)		((val) >> 8)
 #define ELF32_R_TYPE(val)		((val) & 0xff)
 #define ELF32_R_INFO(sym, type)		(((sym) << 8) + ((type) & 0xff))
-- 
2.42.0

